### PR TITLE
Update end to end tests

### PR DESCRIPTION
**Pull request by aarnq**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

Updates end-to-end tests to be usable under the updated test harness.

#### Information to reviewers

I am not able to test the Grafana dashboards properly on a local cluster, but we should probably rethink the strategy here to ensure that we can test it in a more manageable way.

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: tests/end-to-end/harbor/use-api.bats [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
-  load "../bats.lib.bash"
+  load "../../bats.lib.bash"
   load_common "harbor.bash"
   load_common "yq.bash"
   load_assert
 
+  mark.setup
+  mark.punch
+
+  export TESTS_MARKER
```
Comment (simonklb) [inline tests/end-to-end/harbor/use-api.bats:18]: I assume this is to bail early if one of the test fails? Did just a quick search and found `$BATS_TEST_COMPLETED`, have you tried checking that in teardown and skip the rest of the run?

**Comment by aarnq** [comment]
I am a bit stumped on the opa-gatekeeper tests, as I am unable to get them properly running in rootless podman with buildah.
So I think I leave them so this might be merged earlier.

File: tests/end-to-end/harbor/use-api.bats [hunk 2]
```diff
@@ -36,6 +36,12 @@ setup() {
   mark.check
 }
 
+teardown() {
+  if [[ "${BATS_TEST_COMPLETED:-}" == 1 ]]; then
+    mark.punch
```
Comment (simonklb) [inline tests/end-to-end/harbor/use-api.bats:41]: I believe if you just run `skip` here the rest of the tests will be skipped

Comment (simonklb) [reply tests/end-to-end/harbor/use-api.bats:41]: Sorry, thought this was in the setup for some reason.

**Review by simonklb — APPROVED** [review]

**Review by Zash — APPROVED** [review]

**Review by lunkan93 — APPROVED** [review]

**Pull request merged**